<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.KEGG.KGML.KGML_parser</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.KEGG-module.html">Package&nbsp;KEGG</a> ::
        <a href="Bio.KEGG.KGML-module.html">Package&nbsp;KGML</a> ::
        Module&nbsp;KGML_parser
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.KEGG.KGML.KGML_parser-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.KEGG.KGML.KGML_parser-module.html">Module Bio.KEGG.KGML.KGML_parser</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2013 by Leighton Pritchard.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""This module provides classes and functions to parse a KGML pathway map.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">The KGML pathway map is parsed into the object structure defined in</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">KGML_Pathway.py in this module.</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">Classes:</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">    - KGMLParser             Parses KGML file</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">Functions:</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">    - read                   Returns a single Pathway object, using KGMLParser</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">      internally</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt><tt class="py-op">.</tt><tt class="py-name">cElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ElementTree</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">ImportError</tt><tt class="py-op">:</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">xml</tt><tt class="py-op">.</tt><tt class="py-name">etree</tt><tt class="py-op">.</tt><tt class="py-name">ElementTree</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ElementTree</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">StringIO</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Package Bio.KEGG=Bio.KEGG-module.html"><a title="Bio.KEGG" class="py-name" href="#" onclick="return doclink('link-3', 'KEGG', 'link-3');">KEGG</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Package Bio.KEGG.KGML=Bio.KEGG.KGML-module.html"><a title="Bio.KEGG.KGML" class="py-name" href="#" onclick="return doclink('link-4', 'KGML', 'link-4');">KGML</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Module Bio.KEGG.KGML.KGML_pathway=Bio.KEGG.KGML.KGML_pathway-module.html"><a title="Bio.KEGG.KGML.KGML_pathway" class="py-name" href="#" onclick="return doclink('link-5', 'KGML_pathway', 'link-5');">KGML_pathway</a></tt> <tt class="py-keyword">import</tt> <tt class="py-op">*</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L36"></a><tt class="py-lineno"> 36</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_parser-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">debug</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="read-expanded"><a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parses a single KEGG Pathway from given file handle.</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">    Returns a single Pathway object.  There should be one and only</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">    one pathway in each file, but there may well be pathological</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">    examples out there.</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-name">iterator</tt> <tt class="py-op">=</tt> <tt id="link-6" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Method Bio.Blast.NCBIStandalone.BlastErrorParser.parse()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.Blast.NCBIStandalone.BlastParser.parse()=Bio.Blast.NCBIStandalone.BlastParser-class.html#parse,Method Bio.Blast.NCBIStandalone.PSIBlastParser.parse()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Function Bio.Motif.parse()=Bio.Motif-module.html#parse,Method Bio.ParserSupport.AbstractParser.parse()=Bio.ParserSupport.AbstractParser-class.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-6', 'parse', 'link-6');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">debug</tt><tt class="py-op">)</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">        <tt class="py-name">first</tt> <tt class="py-op">=</tt> <tt id="link-7" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.DocSQL.IterationCursor.next()=Bio.DocSQL.IterationCursor-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-7', 'next', 'link-7');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">        <tt class="py-name">first</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">first</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No pathways found in handle"</tt><tt class="py-op">)</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">        <tt class="py-name">second</tt> <tt class="py-op">=</tt> <tt id="link-8" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-8', 'next', 'link-7');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">        <tt class="py-name">second</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">second</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"More than one pathway found in handle"</tt><tt class="py-op">)</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">first</tt> </tt>
</div><a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"> </tt>
<a name="parse"></a><div id="parse-def"><a name="L59"></a><tt class="py-lineno"> 59</tt> <a class="py-toggle" href="#" id="parse-toggle" onclick="return toggle('parse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_parser-module.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">debug</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="parse-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="parse-expanded"><a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-docstring">"""Returns an iterator over Pathway elements.</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">    - handle - file handle to a KGML file for parsing</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">    - debug - integer for amount of debug information to print</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">    This is a generator for the return of multiple Pathway objects.</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">    <tt class="py-comment"># Check handle</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-string">'read'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">            <tt class="py-name">exc_txt</tt> <tt class="py-op">=</tt> <tt class="py-string">"An XML-containing handle or an XML string "</tt> <tt class="py-op">+</tt>\ </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">                <tt class="py-string">"must be provided"</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">Exception</tt><tt class="py-op">(</tt><tt class="py-name">exc_txt</tt><tt class="py-op">)</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">    <tt class="py-comment"># Parse XML and return each Pathway</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">event</tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> \ </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">            <tt class="py-name">ElementTree</tt><tt class="py-op">.</tt><tt class="py-name">iterparse</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-9" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.events()=Bio.Phylo.PhyloXMLIO.Parser-class.html#events,Method Bio.Phylo.PhyloXMLIO.Writer.events()=Bio.Phylo.PhyloXMLIO.Writer-class.html#events"><a title="Bio.Phylo.PhyloXMLIO.Parser.events
Bio.Phylo.PhyloXMLIO.Writer.events" class="py-name" href="#" onclick="return doclink('link-9', 'events', 'link-9');">events</a></tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-string">'start'</tt><tt class="py-op">,</tt> <tt class="py-string">'end'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">event</tt> <tt class="py-op">==</tt> <tt class="py-string">"end"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Variable Bio.SeqIO.AbiIO.tag=Bio.SeqIO.AbiIO-module.html#tag"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-10', 'tag', 'link-10');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"pathway"</tt><tt class="py-op">:</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt id="link-11" class="py-name" targets="Class Bio.KEGG.KGML.KGML_parser.KGMLParser=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html"><a title="Bio.KEGG.KGML.KGML_parser.KGMLParser" class="py-name" href="#" onclick="return doclink('link-11', 'KGMLParser', 'link-11');">KGMLParser</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-12" class="py-name"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-12', 'parse', 'link-6');">parse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Crystal.Crystal.clear()=Bio.Crystal.Crystal-class.html#clear,Method Bio.File._IndexedSeqFileDict.clear()=Bio.File._IndexedSeqFileDict-class.html#clear,Method Bio.Index._InMemoryIndex.clear()=Bio.Index._InMemoryIndex-class.html#clear,Method Bio._py3k._ordereddict.OrderedDict.clear()=Bio._py3k._ordereddict.OrderedDict-class.html#clear"><a title="Bio.Crystal.Crystal.clear
Bio.File._IndexedSeqFileDict.clear
Bio.Index._InMemoryIndex.clear
Bio._py3k._ordereddict.OrderedDict.clear" class="py-name" href="#" onclick="return doclink('link-13', 'clear', 'link-13');">clear</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"> </tt>
<a name="KGMLParser"></a><div id="KGMLParser-def"><a name="L84"></a><tt class="py-lineno"> 84</tt> <a class="py-toggle" href="#" id="KGMLParser-toggle" onclick="return toggle('KGMLParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html">KGMLParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="KGMLParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="KGMLParser-expanded"><a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parses a KGML XML Pathway entry into a Pathway object."""</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"> </tt>
<a name="KGMLParser.__init__"></a><div id="KGMLParser.__init__-def"><a name="L87"></a><tt class="py-lineno"> 87</tt> <a class="py-toggle" href="#" id="KGMLParser.__init__-toggle" onclick="return toggle('KGMLParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="KGMLParser.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="KGMLParser.__init__-expanded"><a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Reaction.entry=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#entry,Function Bio.TogoWS.entry()=Bio.TogoWS-module.html#entry"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-14', 'entry', 'link-14');">entry</a></tt> <tt class="py-op">=</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"> </tt>
<a name="KGMLParser.parse"></a><div id="KGMLParser.parse-def"><a name="L90"></a><tt class="py-lineno"> 90</tt> <a class="py-toggle" href="#" id="KGMLParser.parse-toggle" onclick="return toggle('KGMLParser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="KGMLParser.parse-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="KGMLParser.parse-expanded"><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse the input elements."""</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">_parse_pathway</tt><tt class="py-op">(</tt><tt class="py-param">attrib</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">attrib</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method Bio._py3k._ordereddict.OrderedDict.items()=Bio._py3k._ordereddict.OrderedDict-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-15', 'items', 'link-15');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name" targets="Method Bio.Application.AbstractCommandline.__setattr__()=Bio.Application.AbstractCommandline-class.html#__setattr__,Method Bio.DocSQL.QueryRow.__setattr__()=Bio.DocSQL.QueryRow-class.html#__setattr__"><a title="Bio.Application.AbstractCommandline.__setattr__
Bio.DocSQL.QueryRow.__setattr__" class="py-name" href="#" onclick="return doclink('link-16', '__setattr__', 'link-16');">__setattr__</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
</div><a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">_parse_entry</tt><tt class="py-op">(</tt><tt class="py-param">element</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">            <tt class="py-name">new_entry</tt> <tt class="py-op">=</tt> <tt id="link-17" class="py-name" targets="Class Bio.KEGG.KGML.KGML_pathway.Entry=Bio.KEGG.KGML.KGML_pathway.Entry-class.html"><a title="Bio.KEGG.KGML.KGML_pathway.Entry" class="py-name" href="#" onclick="return doclink('link-17', 'Entry', 'link-17');">Entry</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt id="link-18" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.element=Bio.KEGG.KGML.KGML_pathway.Component-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Entry.element=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Graphics.element=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.element=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.element=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Relation.element=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#element,Variable Bio.Phylo._cdao_owl.element=Bio.Phylo._cdao_owl-module.html#element"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-18', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-19', 'items', 'link-15');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">                <tt class="py-name">new_entry</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="Bio.Application.AbstractCommandline.__setattr__
Bio.DocSQL.QueryRow.__setattr__" class="py-name" href="#" onclick="return doclink('link-20', '__setattr__', 'link-16');">__setattr__</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">subelement</tt> <tt class="py-keyword">in</tt> <tt id="link-21" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-21', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">getchildren</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">subelement</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-22', 'tag', 'link-10');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'graphics'</tt><tt class="py-op">:</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">                    <tt class="py-name">_parse_graphics</tt><tt class="py-op">(</tt><tt class="py-name">subelement</tt><tt class="py-op">,</tt> <tt class="py-name">new_entry</tt><tt class="py-op">)</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">subelement</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-23', 'tag', 'link-10');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'component'</tt><tt class="py-op">:</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">                    <tt class="py-name">_parse_component</tt><tt class="py-op">(</tt><tt class="py-name">subelement</tt><tt class="py-op">,</tt> <tt class="py-name">new_entry</tt><tt class="py-op">)</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Method Bio.Blast.ParseBlastTable.BlastTableRec.add_entry()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#add_entry,Method Bio.KEGG.KGML.KGML_pathway.Pathway.add_entry()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#add_entry"><a title="Bio.Blast.ParseBlastTable.BlastTableRec.add_entry
Bio.KEGG.KGML.KGML_pathway.Pathway.add_entry" class="py-name" href="#" onclick="return doclink('link-24', 'add_entry', 'link-24');">add_entry</a></tt><tt class="py-op">(</tt><tt class="py-name">new_entry</tt><tt class="py-op">)</tt> </tt>
</div><a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">_parse_graphics</tt><tt class="py-op">(</tt><tt class="py-param">element</tt><tt class="py-op">,</tt> <tt class="py-param">entry</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">            <tt class="py-name">new_graphics</tt> <tt class="py-op">=</tt> <tt id="link-25" class="py-name" targets="Package Bio.Graphics=Bio.Graphics-module.html,Class Bio.KEGG.KGML.KGML_pathway.Graphics=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html"><a title="Bio.Graphics
Bio.KEGG.KGML.KGML_pathway.Graphics" class="py-name" href="#" onclick="return doclink('link-25', 'Graphics', 'link-25');">Graphics</a></tt><tt class="py-op">(</tt><tt id="link-26" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-26', 'entry', 'link-14');">entry</a></tt><tt class="py-op">)</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt id="link-27" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-27', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-28', 'items', 'link-15');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">                <tt class="py-name">new_graphics</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.Application.AbstractCommandline.__setattr__
Bio.DocSQL.QueryRow.__setattr__" class="py-name" href="#" onclick="return doclink('link-29', '__setattr__', 'link-16');">__setattr__</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">            <tt id="link-30" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-30', 'entry', 'link-14');">entry</a></tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Entry.add_graphics()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#add_graphics"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.add_graphics" class="py-name" href="#" onclick="return doclink('link-31', 'add_graphics', 'link-31');">add_graphics</a></tt><tt class="py-op">(</tt><tt class="py-name">new_graphics</tt><tt class="py-op">)</tt> </tt>
</div><a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">_parse_component</tt><tt class="py-op">(</tt><tt class="py-param">element</tt><tt class="py-op">,</tt> <tt class="py-param">entry</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-name">new_component</tt> <tt class="py-op">=</tt> <tt id="link-32" class="py-name" targets="Class Bio.KEGG.KGML.KGML_pathway.Component=Bio.KEGG.KGML.KGML_pathway.Component-class.html"><a title="Bio.KEGG.KGML.KGML_pathway.Component" class="py-name" href="#" onclick="return doclink('link-32', 'Component', 'link-32');">Component</a></tt><tt class="py-op">(</tt><tt id="link-33" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-33', 'entry', 'link-14');">entry</a></tt><tt class="py-op">)</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt id="link-34" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-34', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-35', 'items', 'link-15');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">                <tt class="py-name">new_component</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.Application.AbstractCommandline.__setattr__
Bio.DocSQL.QueryRow.__setattr__" class="py-name" href="#" onclick="return doclink('link-36', '__setattr__', 'link-16');">__setattr__</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">            <tt id="link-37" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-37', 'entry', 'link-14');">entry</a></tt><tt class="py-op">.</tt><tt id="link-38" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Entry.add_component()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#add_component"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.add_component" class="py-name" href="#" onclick="return doclink('link-38', 'add_component', 'link-38');">add_component</a></tt><tt class="py-op">(</tt><tt class="py-name">new_component</tt><tt class="py-op">)</tt> </tt>
</div><a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">_parse_reaction</tt><tt class="py-op">(</tt><tt class="py-param">element</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">            <tt class="py-name">new_reaction</tt> <tt class="py-op">=</tt> <tt id="link-39" class="py-name" targets="Class Bio.KEGG.KGML.KGML_pathway.Reaction=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html,Class Bio.Pathway.Reaction=Bio.Pathway.Reaction-class.html"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction
Bio.Pathway.Reaction" class="py-name" href="#" onclick="return doclink('link-39', 'Reaction', 'link-39');">Reaction</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt id="link-40" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-40', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-41', 'items', 'link-15');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">                <tt class="py-name">new_reaction</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Application.AbstractCommandline.__setattr__
Bio.DocSQL.QueryRow.__setattr__" class="py-name" href="#" onclick="return doclink('link-42', '__setattr__', 'link-16');">__setattr__</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">subelement</tt> <tt class="py-keyword">in</tt> <tt id="link-43" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-43', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">getchildren</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">subelement</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-44', 'tag', 'link-10');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'substrate'</tt><tt class="py-op">:</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">                    <tt class="py-name">new_reaction</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Reaction.add_substrate()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#add_substrate"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.add_substrate" class="py-name" href="#" onclick="return doclink('link-45', 'add_substrate', 'link-45');">add_substrate</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">subelement</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">subelement</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-46', 'tag', 'link-10');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'product'</tt><tt class="py-op">:</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">                    <tt class="py-name">new_reaction</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Reaction.add_product()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#add_product"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.add_product" class="py-name" href="#" onclick="return doclink('link-47', 'add_product', 'link-47');">add_product</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">subelement</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Pathway.add_reaction()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#add_reaction,Method Bio.Pathway.System.add_reaction()=Bio.Pathway.System-class.html#add_reaction"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.add_reaction
Bio.Pathway.System.add_reaction" class="py-name" href="#" onclick="return doclink('link-48', 'add_reaction', 'link-48');">add_reaction</a></tt><tt class="py-op">(</tt><tt class="py-name">new_reaction</tt><tt class="py-op">)</tt> </tt>
</div><a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">_parse_relation</tt><tt class="py-op">(</tt><tt class="py-param">element</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">            <tt class="py-name">new_relation</tt> <tt class="py-op">=</tt> <tt id="link-49" class="py-name" targets="Class Bio.KEGG.KGML.KGML_pathway.Relation=Bio.KEGG.KGML.KGML_pathway.Relation-class.html"><a title="Bio.KEGG.KGML.KGML_pathway.Relation" class="py-name" href="#" onclick="return doclink('link-49', 'Relation', 'link-49');">Relation</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">            <tt class="py-name">new_relation</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Relation.entry1=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#entry1"><a title="Bio.KEGG.KGML.KGML_pathway.Relation.entry1" class="py-name" href="#" onclick="return doclink('link-50', 'entry1', 'link-50');">entry1</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-51" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-51', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'entry1'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">            <tt class="py-name">new_relation</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Relation.entry2=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#entry2"><a title="Bio.KEGG.KGML.KGML_pathway.Relation.entry2" class="py-name" href="#" onclick="return doclink('link-52', 'entry2', 'link-52');">entry2</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-53" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-53', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'entry2'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">            <tt class="py-name">new_relation</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-54', 'type', 'link-54');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-55" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-55', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'type'</tt><tt class="py-op">]</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">subtype</tt> <tt class="py-keyword">in</tt> <tt id="link-56" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-56', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt class="py-name">getchildren</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">                <tt id="link-57" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-57', 'name', 'link-57');">name</a></tt><tt class="py-op">,</tt> <tt id="link-58" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-58', 'value', 'link-58');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">subtype</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'name'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">subtype</tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">[</tt><tt class="py-string">'value'</tt><tt class="py-op">]</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-59" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-59', 'name', 'link-57');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'compound'</tt><tt class="py-op">,</tt> <tt class="py-string">'hidden compound'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">                    <tt class="py-name">new_relation</tt><tt class="py-op">.</tt><tt class="py-name">subtypes</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-60', 'append', 'link-60');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-61" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-61', 'name', 'link-57');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-62', 'value', 'link-58');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">                    <tt class="py-name">new_relation</tt><tt class="py-op">.</tt><tt class="py-name">subtypes</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-63', 'append', 'link-60');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-64" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-64', 'name', 'link-57');">name</a></tt><tt class="py-op">,</tt> <tt id="link-65" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-65', 'value', 'link-58');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Pathway.add_relation()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#add_relation"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.add_relation" class="py-name" href="#" onclick="return doclink('link-66', 'add_relation', 'link-66');">add_relation</a></tt><tt class="py-op">(</tt><tt class="py-name">new_relation</tt><tt class="py-op">)</tt> </tt>
</div><a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">        <tt class="py-comment"># ==========</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">        <tt class="py-comment"># Initialise Pathway</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pathway</tt> <tt class="py-op">=</tt> <tt id="link-67" class="py-name" targets="Class Bio.KEGG.KGML.KGML_pathway.Pathway=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html,Package Bio.Pathway=Bio.Pathway-module.html"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway
Bio.Pathway" class="py-name" href="#" onclick="return doclink('link-67', 'Pathway', 'link-67');">Pathway</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">        <tt class="py-comment"># Get information about the pathway itself</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-name">_parse_pathway</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-68', 'entry', 'link-14');">entry</a></tt><tt class="py-op">.</tt><tt class="py-name">attrib</tt><tt class="py-op">)</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-69" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-69', 'element', 'link-18');">element</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-70', 'entry', 'link-14');">entry</a></tt><tt class="py-op">.</tt><tt class="py-name">getchildren</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-71" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-71', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-72', 'tag', 'link-10');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'entry'</tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">                <tt class="py-name">_parse_entry</tt><tt class="py-op">(</tt><tt id="link-73" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-73', 'element', 'link-18');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-74" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-74', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-75', 'tag', 'link-10');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'reaction'</tt><tt class="py-op">:</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">                <tt class="py-name">_parse_reaction</tt><tt class="py-op">(</tt><tt id="link-76" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-76', 'element', 'link-18');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-77" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-77', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-78', 'tag', 'link-10');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'relation'</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">                <tt class="py-name">_parse_relation</tt><tt class="py-op">(</tt><tt id="link-79" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-79', 'element', 'link-18');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">            <tt class="py-comment"># Parsing of some elements not implemented - no examples yet</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                <tt class="py-comment"># This should warn us of any unimplemented tags</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">                <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                <tt class="py-keyword">from</tt> <tt id="link-80" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-80', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-81" class="py-name" targets="Class Bio.BiopythonParserWarning=Bio.BiopythonParserWarning-class.html"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-81', 'BiopythonParserWarning', 'link-81');">BiopythonParserWarning</a></tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Warning: tag %s not implemented in parser"</tt> <tt class="py-op">%</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                              <tt id="link-82" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-82', 'element', 'link-18');">element</a></tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-83', 'tag', 'link-10');">tag</a></tt><tt class="py-op">,</tt> <tt id="link-84" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-84', 'BiopythonParserWarning', 'link-81');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pathway</tt> </tt>
</div></div><a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line"> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">'__main__'</tt><tt class="py-op">:</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">    <tt class="py-comment"># Check large metabolism</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">    <tt class="py-name">pathway</tt> <tt class="py-op">=</tt> <tt id="link-85" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Function Bio.Motif.Parsers.AlignAce.read()=Bio.Motif.Parsers.AlignAce-module.html#read,Function Bio.Motif.Parsers.MAST.read()=Bio.Motif.Parsers.MAST-module.html#read,Function Bio.Motif.Parsers.MEME.read()=Bio.Motif.Parsers.MEME-module.html#read,Function Bio.Motif.read()=Bio.Motif-module.html#read,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.read()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-85', 'read', 'link-85');">read</a></tt><tt class="py-op">(</tt><tt id="link-86" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-86', 'open', 'link-86');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">'ko01100.xml'</tt><tt class="py-op">,</tt> <tt class="py-string">'rU'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">)</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-87', 'items', 'link-15');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">20</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.reactions=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#reactions,Method Bio.Pathway.System.reactions()=Bio.Pathway.System-class.html#reactions"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.reactions
Bio.Pathway.System.reactions" class="py-name" href="#" onclick="return doclink('link-88', 'reactions', 'link-88');">reactions</a></tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">20</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">r</tt><tt class="py-op">)</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.maps=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#maps"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.maps" class="py-name" href="#" onclick="return doclink('link-89', 'maps', 'link-89');">maps</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">    <tt class="py-comment"># Check relations</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">    <tt class="py-name">pathway</tt> <tt class="py-op">=</tt> <tt id="link-90" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-90', 'read', 'link-85');">read</a></tt><tt class="py-op">(</tt><tt id="link-91" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-91', 'open', 'link-86');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">'ko_metabolic/ko00010.xml'</tt><tt class="py-op">,</tt> <tt class="py-string">'rU'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">)</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-92', 'items', 'link-15');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">20</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.reactions
Bio.Pathway.System.reactions" class="py-name" href="#" onclick="return doclink('link-93', 'reactions', 'link-88');">reactions</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">20</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">r</tt><tt class="py-op">)</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.relations=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#relations"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.relations" class="py-name" href="#" onclick="return doclink('link-94', 'relations', 'link-94');">relations</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">20</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">r</tt><tt class="py-op">)</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.maps" class="py-name" href="#" onclick="return doclink('link-95', 'maps', 'link-89');">maps</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">    <tt class="py-comment"># Check components</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">    <tt class="py-name">pathway</tt> <tt class="py-op">=</tt> <tt id="link-96" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-96', 'read', 'link-85');">read</a></tt><tt class="py-op">(</tt><tt id="link-97" class="py-name"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-97', 'open', 'link-86');">open</a></tt><tt class="py-op">(</tt><tt class="py-string">'ko_metabolic/ko00253.xml'</tt><tt class="py-op">,</tt> <tt class="py-string">'rU'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">)</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt class="py-name">entries</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-98', 'items', 'link-15');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.maps" class="py-name" href="#" onclick="return doclink('link-99', 'maps', 'link-89');">maps</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">    <tt class="py-comment"># Test XML representation</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name" targets="Method Bio.KEGG.KGML.KGML_pathway.Pathway.get_KGML()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#get_KGML"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.get_KGML" class="py-name" href="#" onclick="return doclink('link-100', 'get_KGML', 'link-100');">get_KGML</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line"> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">    <tt class="py-comment"># Test bounds of pathway</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">pathway</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.bounds=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#bounds,Variable Bio.KEGG.KGML.KGML_pathway.Graphics.bounds=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#bounds,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.bounds=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#bounds"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.bounds
Bio.KEGG.KGML.KGML_pathway.Graphics.bounds
Bio.KEGG.KGML.KGML_pathway.Pathway.bounds" class="py-name" href="#" onclick="return doclink('link-101', 'bounds', 'link-101');">bounds</a></tt><tt class="py-op">)</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:05:07 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
