<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.PopGen.FDist.Utils</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.PopGen-module.html">Package&nbsp;PopGen</a> ::
        <a href="Bio.PopGen.FDist-module.html">Package&nbsp;FDist</a> ::
        Module&nbsp;Utils
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.PopGen.FDist.Utils-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.PopGen.FDist.Utils-module.html">Module Bio.PopGen.FDist.Utils</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2007 by Tiago Antao &lt;tiagoantao@gmail.com&gt;.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.PopGen=Bio.PopGen-module.html"><a title="Bio.PopGen" class="py-name" href="#" onclick="return doclink('link-1', 'PopGen', 'link-1');">PopGen</a></tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Package Bio.PopGen.GenePop=Bio.PopGen.GenePop-module.html"><a title="Bio.PopGen.GenePop" class="py-name" href="#" onclick="return doclink('link-2', 'GenePop', 'link-2');">GenePop</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Module Bio.PopGen.GenePop.FileParser=Bio.PopGen.GenePop.FileParser-module.html"><a title="Bio.PopGen.GenePop.FileParser" class="py-name" href="#" onclick="return doclink('link-3', 'FileParser', 'link-3');">FileParser</a></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name"><a title="Bio.PopGen" class="py-name" href="#" onclick="return doclink('link-5', 'PopGen', 'link-1');">PopGen</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Package Bio.PopGen.FDist=Bio.PopGen.FDist-module.html"><a title="Bio.PopGen.FDist" class="py-name" href="#" onclick="return doclink('link-6', 'FDist', 'link-6');">FDist</a></tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment"># Quite a few utility functions could be done (like remove pop,</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-comment"># add locus, etc...). The recommended strategy is convert back</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-comment"># and forth from/to GenePop and use GenePop Utils</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="convert_genepop_to_fdist"></a><div id="convert_genepop_to_fdist-def"><a name="L15"></a><tt class="py-lineno"> 15</tt> <a class="py-toggle" href="#" id="convert_genepop_to_fdist-toggle" onclick="return toggle('convert_genepop_to_fdist');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PopGen.FDist.Utils-module.html#convert_genepop_to_fdist">convert_genepop_to_fdist</a><tt class="py-op">(</tt><tt class="py-param">gp_rec</tt><tt class="py-op">,</tt> <tt class="py-param">report_pops</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="convert_genepop_to_fdist-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="convert_genepop_to_fdist-expanded"><a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line">    <tt class="py-docstring">"""Converts a GenePop record to a FDist one.</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">       Parameters:</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">       gp_rec - Genepop Record (either standard or big)</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">       Returns:</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">       FDist record.</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">,</tt> <tt class="py-string">"populations"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-7" class="py-name" targets="Function Bio.PopGen.FDist.Utils._convert_genepop_to_fdist()=Bio.PopGen.FDist.Utils-module.html#_convert_genepop_to_fdist"><a title="Bio.PopGen.FDist.Utils._convert_genepop_to_fdist" class="py-name" href="#" onclick="return doclink('link-7', '_convert_genepop_to_fdist', 'link-7');">_convert_genepop_to_fdist</a></tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">)</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-8" class="py-name" targets="Function Bio.PopGen.FDist.Utils._convert_genepop_to_fdist_big()=Bio.PopGen.FDist.Utils-module.html#_convert_genepop_to_fdist_big"><a title="Bio.PopGen.FDist.Utils._convert_genepop_to_fdist_big" class="py-name" href="#" onclick="return doclink('link-8', '_convert_genepop_to_fdist_big', 'link-8');">_convert_genepop_to_fdist_big</a></tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">,</tt> <tt class="py-name">report_pops</tt><tt class="py-op">)</tt> </tt>
</div><a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="_convert_genepop_to_fdist"></a><div id="_convert_genepop_to_fdist-def"><a name="L30"></a><tt class="py-lineno"> 30</tt> <a class="py-toggle" href="#" id="_convert_genepop_to_fdist-toggle" onclick="return toggle('_convert_genepop_to_fdist');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PopGen.FDist.Utils-module.html#_convert_genepop_to_fdist">_convert_genepop_to_fdist</a><tt class="py-op">(</tt><tt class="py-param">gp_rec</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_convert_genepop_to_fdist-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_convert_genepop_to_fdist-expanded"><a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">    <tt class="py-docstring">"""Converts a standard GenePop record to a FDist one.</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">       Parameters:</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">       gp_rec - Genepop Record (Standard)</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">       Returns:</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">       FDist record.</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt> <tt class="py-op">=</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name"><a title="Bio.PopGen" class="py-name" href="#" onclick="return doclink('link-10', 'PopGen', 'link-1');">PopGen</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name"><a title="Bio.PopGen.FDist" class="py-name" href="#" onclick="return doclink('link-11', 'FDist', 'link-6');">FDist</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.Motif.Parsers.AlignAce.Record=Bio.Motif.Parsers.AlignAce.Record-class.html,Class Bio.Motif.Parsers.MAST.Record=Bio.Motif.Parsers.MAST.Record-class.html,Class Bio.PopGen.FDist.Record=Bio.PopGen.FDist.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-12', 'Record', 'link-12');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">data_org</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">num_loci</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_list</tt><tt class="py-op">)</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">num_pops</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">populations</tt><tt class="py-op">)</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">lc_i</tt> <tt class="py-keyword">in</tt> <tt id="link-13" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-13', 'range', 'link-13');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">        <tt class="py-name">alleles</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">        <tt class="py-name">pop_data</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">pop_i</tt> <tt class="py-keyword">in</tt> <tt id="link-14" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-14', 'range', 'link-13');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">populations</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">indiv</tt> <tt class="py-keyword">in</tt> <tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">populations</tt><tt class="py-op">[</tt><tt class="py-name">pop_i</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">al</tt> <tt class="py-keyword">in</tt> <tt class="py-name">indiv</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">lc_i</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">al</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">al</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alleles</tt><tt class="py-op">:</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">                        <tt class="py-name">alleles</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-15', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">al</tt><tt class="py-op">)</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">        <tt class="py-name">alleles</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.sort()=Bio.Align.MultipleSeqAlignment-class.html#sort,Method Bio.PDB.Residue.DisorderedResidue.sort()=Bio.PDB.Residue.DisorderedResidue-class.html#sort,Method Bio.PDB.Residue.Residue.sort()=Bio.PDB.Residue.Residue-class.html#sort,Method Bio.SearchIO._model.hit.Hit.sort()=Bio.SearchIO._model.hit.Hit-class.html#sort,Method Bio.SearchIO._model.query.QueryResult.sort()=Bio.SearchIO._model.query.QueryResult-class.html#sort,Method Bio.Sequencing.Ace.ACEFileRecord.sort()=Bio.Sequencing.Ace.ACEFileRecord-class.html#sort"><a title="Bio.Align.MultipleSeqAlignment.sort
Bio.PDB.Residue.DisorderedResidue.sort
Bio.PDB.Residue.Residue.sort
Bio.SearchIO._model.hit.Hit.sort
Bio.SearchIO._model.query.QueryResult.sort
Bio.Sequencing.Ace.ACEFileRecord.sort" class="py-name" href="#" onclick="return doclink('link-16', 'sort', 'link-16');">sort</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Dominance requires this</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-comment"># here we go again (necessary...)</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">pop_i</tt> <tt class="py-keyword">in</tt> <tt id="link-17" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-17', 'range', 'link-13');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">populations</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">            <tt class="py-name">allele_counts</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">indiv</tt> <tt class="py-keyword">in</tt> <tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">populations</tt><tt class="py-op">[</tt><tt class="py-name">pop_i</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">al</tt> <tt class="py-keyword">in</tt> <tt class="py-name">indiv</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">lc_i</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">al</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">                        <tt id="link-18" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-18', 'count', 'link-18');">count</a></tt> <tt class="py-op">=</tt> <tt class="py-name">allele_counts</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-19', 'get', 'link-19');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">al</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">                        <tt class="py-name">allele_counts</tt><tt class="py-op">[</tt><tt class="py-name">al</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-20" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-20', 'count', 'link-18');">count</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">            <tt class="py-name">allele_array</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># We need the same order as in alleles</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">allele</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alleles</tt><tt class="py-op">:</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">                <tt class="py-name">allele_array</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-21', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">allele_counts</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-22', 'get', 'link-19');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">allele</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">            <tt class="py-name">pop_data</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-23', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">allele_array</tt><tt class="py-op">)</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_data</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-24', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">alleles</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pop_data</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">fd_rec</tt> </tt>
</div><a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"> </tt>
<a name="_convert_genepop_to_fdist_big"></a><div id="_convert_genepop_to_fdist_big-def"><a name="L70"></a><tt class="py-lineno"> 70</tt> <a class="py-toggle" href="#" id="_convert_genepop_to_fdist_big-toggle" onclick="return toggle('_convert_genepop_to_fdist_big');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PopGen.FDist.Utils-module.html#_convert_genepop_to_fdist_big">_convert_genepop_to_fdist_big</a><tt class="py-op">(</tt><tt class="py-param">gp_rec</tt><tt class="py-op">,</tt> <tt class="py-param">report_pops</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_convert_genepop_to_fdist_big-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_convert_genepop_to_fdist_big-expanded"><a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">    <tt class="py-docstring">"""Converts a big GenePop record to a FDist one.</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring">       Parameters:</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">       gp_rec - Genepop Record (Big)</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">       Returns:</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">       FDist record.</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt> <tt class="py-op">=</tt> <tt id="link-25" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-25', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.PopGen" class="py-name" href="#" onclick="return doclink('link-26', 'PopGen', 'link-1');">PopGen</a></tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.PopGen.FDist" class="py-name" href="#" onclick="return doclink('link-27', 'FDist', 'link-6');">FDist</a></tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-28', 'Record', 'link-12');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">data_org</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">num_loci</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_list</tt><tt class="py-op">)</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">    <tt class="py-name">num_loci</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_list</tt><tt class="py-op">)</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">    <tt class="py-name">loci</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-29" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-29', 'i', 'link-29');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-30" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-30', 'range', 'link-13');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">num_loci</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">        <tt class="py-name">loci</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-31', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt id="link-32" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-32', 'set', 'link-32');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-name">pops</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">    <tt class="py-name">work_rec</tt> <tt class="py-op">=</tt> <tt id="link-33" class="py-name"><a title="Bio.PopGen.GenePop.FileParser" class="py-name" href="#" onclick="return doclink('link-33', 'FileParser', 'link-3');">FileParser</a></tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Function Bio.Motif.Parsers.AlignAce.read()=Bio.Motif.Parsers.AlignAce-module.html#read,Function Bio.Motif.Parsers.MAST.read()=Bio.Motif.Parsers.MAST-module.html#read,Function Bio.Motif.Parsers.MEME.read()=Bio.Motif.Parsers.MEME-module.html#read,Function Bio.Motif.read()=Bio.Motif-module.html#read,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.read()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-34', 'read', 'link-34');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">fname</tt><tt class="py-op">)</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">    <tt class="py-name">lParser</tt> <tt class="py-op">=</tt> <tt class="py-name">work_rec</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name" targets="Method Bio.PopGen.GenePop.FileParser.FileRecord.get_individual()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#get_individual"><a title="Bio.PopGen.GenePop.FileParser.FileRecord.get_individual" class="py-name" href="#" onclick="return doclink('link-35', 'get_individual', 'link-35');">get_individual</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">init_pop</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-name">my_pop</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-36" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-36', 'i', 'link-29');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-37" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-37', 'range', 'link-13');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">num_loci</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">            <tt class="py-name">my_pop</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-38', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">my_pop</tt> </tt>
</div><a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-name">curr_pop</tt> <tt class="py-op">=</tt> <tt class="py-name">init_pop</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-name">num_pops</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">report_pops</tt><tt class="py-op">:</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-name">report_pops</tt><tt class="py-op">(</tt><tt class="py-name">num_pops</tt><tt class="py-op">)</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">lParser</tt><tt class="py-op">:</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">lParser</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">loci_pos</tt> <tt class="py-keyword">in</tt> <tt id="link-39" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-39', 'range', 'link-13');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">num_loci</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">al</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lParser</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">loci_pos</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">al</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">                        <tt class="py-name">loci</tt><tt class="py-op">[</tt><tt class="py-name">loci_pos</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Method Bio.Graphics.BasicChromosome._ChromosomeComponent.add()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#add,Method Bio.Nexus.Nexus.StepMatrix.add()=Bio.Nexus.Nexus.StepMatrix-class.html#add,Method Bio.Nexus.Nodes.Chain.add()=Bio.Nexus.Nodes.Chain-class.html#add,Method Bio.PDB.Entity.Entity.add()=Bio.PDB.Entity.Entity-class.html#add,Method Bio.PDB.Residue.DisorderedResidue.add()=Bio.PDB.Residue.DisorderedResidue-class.html#add,Method Bio.PDB.Residue.Residue.add()=Bio.PDB.Residue.Residue-class.html#add,Method Bio.Restriction.Restriction.RestrictionBatch.add()=Bio.Restriction.Restriction.RestrictionBatch-class.html#add"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-40', 'add', 'link-40');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">al</tt><tt class="py-op">)</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">                        <tt class="py-name">curr_pop</tt><tt class="py-op">[</tt><tt class="py-name">loci_pos</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">al</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">curr_pop</tt><tt class="py-op">[</tt><tt class="py-name">loci_pos</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-41', 'get', 'link-19');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">al</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">            <tt class="py-name">pops</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-42', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">curr_pop</tt><tt class="py-op">)</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt class="py-name">num_pops</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">report_pops</tt><tt class="py-op">:</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">                <tt class="py-name">report_pops</tt><tt class="py-op">(</tt><tt class="py-name">num_pops</tt><tt class="py-op">)</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-name">curr_pop</tt> <tt class="py-op">=</tt> <tt class="py-name">init_pop</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-name">lParser</tt> <tt class="py-op">=</tt> <tt class="py-name">work_rec</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.PopGen.GenePop.FileParser.FileRecord.get_individual" class="py-name" href="#" onclick="return doclink('link-43', 'get_individual', 'link-35');">get_individual</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">    <tt class="py-name">work_rec</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.close()=Bio.File._IndexedSeqFileDict-class.html#close,Method Bio.File._SQLiteManySeqFilesDict.close()=Bio.File._SQLiteManySeqFilesDict-class.html#close,Method Bio.SeqIO.SffIO._AddTellHandle.close()=Bio.SeqIO.SffIO._AddTellHandle-class.html#close,Method Bio.bgzf.BgzfReader.close()=Bio.bgzf.BgzfReader-class.html#close,Method Bio.bgzf.BgzfWriter.close()=Bio.bgzf.BgzfWriter-class.html#close,Method BioSQL.BioSeqDatabase.Adaptor.close()=BioSQL.BioSeqDatabase.Adaptor-class.html#close,Method BioSQL.BioSeqDatabase.DBServer.close()=BioSQL.BioSeqDatabase.DBServer-class.html#close"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-44', 'close', 'link-44');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  <tt class="py-comment"># TODO - Needs a proper fix</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">    <tt class="py-name">pops</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-45', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">curr_pop</tt><tt class="py-op">)</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">num_pops</tt> <tt class="py-op">=</tt> <tt class="py-name">num_pops</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">loci_pos</tt> <tt class="py-keyword">in</tt> <tt id="link-46" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-46', 'range', 'link-13');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">num_loci</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-name">alleles</tt> <tt class="py-op">=</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">loci</tt><tt class="py-op">[</tt><tt class="py-name">loci_pos</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-name">loci_rec</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">alleles</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-47" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop,Method Bio._py3k._ordereddict.OrderedDict.pop()=Bio._py3k._ordereddict.OrderedDict-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-47', 'pop', 'link-47');">pop</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">pops</tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">            <tt class="py-name">pop_rec</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">allele</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alleles</tt><tt class="py-op">:</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">                <tt class="py-name">pop_rec</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-48', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt id="link-49" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-49', 'pop', 'link-47');">pop</a></tt><tt class="py-op">[</tt><tt class="py-name">loci_pos</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-50', 'get', 'link-19');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">allele</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">            <tt class="py-name">loci_rec</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-51', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">pop_rec</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">        <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_data</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-52', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">loci_rec</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">fd_rec</tt> </tt>
</div><a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"> </tt>
<a name="_convert_genepop_to_fdist_big_old"></a><div id="_convert_genepop_to_fdist_big_old-def"><a name="L130"></a><tt class="py-lineno">130</tt> <a class="py-toggle" href="#" id="_convert_genepop_to_fdist_big_old-toggle" onclick="return toggle('_convert_genepop_to_fdist_big_old');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PopGen.FDist.Utils-module.html#_convert_genepop_to_fdist_big_old">_convert_genepop_to_fdist_big_old</a><tt class="py-op">(</tt><tt class="py-param">gp_rec</tt><tt class="py-op">,</tt> <tt class="py-param">report_loci</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_convert_genepop_to_fdist_big_old-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_convert_genepop_to_fdist_big_old-expanded"><a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">    <tt class="py-docstring">"""Converts a big GenePop record to a FDist one.</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">       Parameters:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring">       gp_rec - Genepop Record (Big)</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"><tt class="py-docstring">       Returns:</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"><tt class="py-docstring">       FDist record.</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt> <tt class="py-op">=</tt> <tt id="link-53" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-53', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.PopGen" class="py-name" href="#" onclick="return doclink('link-54', 'PopGen', 'link-1');">PopGen</a></tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.PopGen.FDist" class="py-name" href="#" onclick="return doclink('link-55', 'FDist', 'link-6');">FDist</a></tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-56', 'Record', 'link-12');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">countPops</tt><tt class="py-op">(</tt><tt class="py-param">rec</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">        <tt class="py-name">f2</tt> <tt class="py-op">=</tt> <tt id="link-57" class="py-name"><a title="Bio.PopGen.GenePop.FileParser" class="py-name" href="#" onclick="return doclink('link-57', 'FileParser', 'link-3');">FileParser</a></tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-58', 'read', 'link-34');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">rec</tt><tt class="py-op">.</tt><tt class="py-name">fname</tt><tt class="py-op">)</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">        <tt class="py-name">popCnt</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">f2</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Method Bio.PopGen.GenePop.FileParser.FileRecord.skip_population()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#skip_population"><a title="Bio.PopGen.GenePop.FileParser.FileRecord.skip_population" class="py-name" href="#" onclick="return doclink('link-59', 'skip_population', 'link-59');">skip_population</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">            <tt class="py-name">popCnt</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">popCnt</tt> </tt>
</div><a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">data_org</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">num_loci</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_list</tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-name">work_rec0</tt> <tt class="py-op">=</tt> <tt id="link-60" class="py-name"><a title="Bio.PopGen.GenePop.FileParser" class="py-name" href="#" onclick="return doclink('link-60', 'FileParser', 'link-3');">FileParser</a></tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-61', 'read', 'link-34');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">fname</tt><tt class="py-op">)</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">    <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">num_pops</tt> <tt class="py-op">=</tt> <tt class="py-name">countPops</tt><tt class="py-op">(</tt><tt class="py-name">work_rec0</tt><tt class="py-op">)</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">    <tt class="py-name">num_loci</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_list</tt><tt class="py-op">)</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">lc_i</tt> <tt class="py-keyword">in</tt> <tt id="link-62" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-62', 'range', 'link-13');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">num_loci</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">report_loci</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">            <tt class="py-name">report_loci</tt><tt class="py-op">(</tt><tt class="py-name">lc_i</tt><tt class="py-op">,</tt> <tt class="py-name">num_loci</tt><tt class="py-op">)</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">        <tt class="py-name">work_rec</tt> <tt class="py-op">=</tt> <tt id="link-63" class="py-name"><a title="Bio.PopGen.GenePop.FileParser" class="py-name" href="#" onclick="return doclink('link-63', 'FileParser', 'link-3');">FileParser</a></tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-64', 'read', 'link-34');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">fname</tt><tt class="py-op">)</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-name">work_rec2</tt> <tt class="py-op">=</tt> <tt id="link-65" class="py-name"><a title="Bio.PopGen.GenePop.FileParser" class="py-name" href="#" onclick="return doclink('link-65', 'FileParser', 'link-3');">FileParser</a></tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-66', 'read', 'link-34');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">gp_rec</tt><tt class="py-op">.</tt><tt class="py-name">fname</tt><tt class="py-op">)</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-name">alleles</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-name">pop_data</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-name">lParser</tt> <tt class="py-op">=</tt> <tt class="py-name">work_rec</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.PopGen.GenePop.FileParser.FileRecord.get_individual" class="py-name" href="#" onclick="return doclink('link-67', 'get_individual', 'link-35');">get_individual</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">lParser</tt><tt class="py-op">:</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">lParser</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">al</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lParser</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">lc_i</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">al</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">al</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alleles</tt><tt class="py-op">:</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">                        <tt class="py-name">alleles</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-68', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">al</tt><tt class="py-op">)</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">            <tt class="py-name">lParser</tt> <tt class="py-op">=</tt> <tt class="py-name">work_rec</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.PopGen.GenePop.FileParser.FileRecord.get_individual" class="py-name" href="#" onclick="return doclink('link-69', 'get_individual', 'link-35');">get_individual</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-comment"># here we go again (necessary...)</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-name">alleles</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.sort
Bio.PDB.Residue.DisorderedResidue.sort
Bio.PDB.Residue.Residue.sort
Bio.SearchIO._model.hit.Hit.sort
Bio.SearchIO._model.query.QueryResult.sort
Bio.Sequencing.Ace.ACEFileRecord.sort" class="py-name" href="#" onclick="return doclink('link-70', 'sort', 'link-16');">sort</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">process_pop</tt><tt class="py-op">(</tt><tt class="py-param">pop_data</tt><tt class="py-op">,</tt> <tt class="py-param">alleles</tt><tt class="py-op">,</tt> <tt class="py-param">allele_counts</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">            <tt class="py-name">allele_array</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># We need the same order as in alleles</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">allele</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alleles</tt><tt class="py-op">:</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">                <tt class="py-name">allele_array</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-71', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">allele_counts</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-72', 'get', 'link-19');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">allele</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt class="py-name">pop_data</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-73', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">allele_array</tt><tt class="py-op">)</tt> </tt>
</div><a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-name">lParser</tt> <tt class="py-op">=</tt> <tt class="py-name">work_rec2</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="Bio.PopGen.GenePop.FileParser.FileRecord.get_individual" class="py-name" href="#" onclick="return doclink('link-74', 'get_individual', 'link-35');">get_individual</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">        <tt class="py-name">allele_counts</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">allele</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alleles</tt><tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">            <tt class="py-name">allele_counts</tt><tt class="py-op">[</tt><tt class="py-name">allele</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">        <tt class="py-name">allele_counts</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">lParser</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">lParser</tt> <tt class="py-keyword">is</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">                <tt class="py-name">process_pop</tt><tt class="py-op">(</tt><tt class="py-name">pop_data</tt><tt class="py-op">,</tt> <tt class="py-name">alleles</tt><tt class="py-op">,</tt> <tt class="py-name">allele_counts</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">                <tt class="py-name">allele_counts</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">allele</tt> <tt class="py-keyword">in</tt> <tt class="py-name">alleles</tt><tt class="py-op">:</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">                    <tt class="py-name">allele_counts</tt><tt class="py-op">[</tt><tt class="py-name">allele</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">                <tt class="py-name">allele_counts</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">al</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lParser</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">lc_i</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">                    <tt class="py-name">allele_counts</tt><tt class="py-op">[</tt><tt class="py-name">al</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">            <tt class="py-name">lParser</tt> <tt class="py-op">=</tt> <tt class="py-name">work_rec2</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.PopGen.GenePop.FileParser.FileRecord.get_individual" class="py-name" href="#" onclick="return doclink('link-75', 'get_individual', 'link-35');">get_individual</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-name">process_pop</tt><tt class="py-op">(</tt><tt class="py-name">pop_data</tt><tt class="py-op">,</tt> <tt class="py-name">alleles</tt><tt class="py-op">,</tt> <tt class="py-name">allele_counts</tt><tt class="py-op">)</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-name">fd_rec</tt><tt class="py-op">.</tt><tt class="py-name">loci_data</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-76', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">alleles</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pop_data</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">fd_rec</tt> </tt>
</div><a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"> </tt>
<a name="approximate_fst"></a><div id="approximate_fst-def"><a name="L199"></a><tt class="py-lineno">199</tt> <a class="py-toggle" href="#" id="approximate_fst-toggle" onclick="return toggle('approximate_fst');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PopGen.FDist.Utils-module.html#approximate_fst">approximate_fst</a><tt class="py-op">(</tt><tt class="py-param">desired_fst</tt><tt class="py-op">,</tt> <tt class="py-param">simulated_fst</tt><tt class="py-op">,</tt> <tt class="py-param">parameter_fst</tt><tt class="py-op">,</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">           <tt class="py-param">max_run_fst</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-param">min_run_fst</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">limit</tt><tt class="py-op">=</tt><tt class="py-number">0.005</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="approximate_fst-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="approximate_fst-expanded"><a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">    <tt class="py-docstring">"""Calculates the next Fst attempt in order to approximate a</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring">       desired Fst.</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-77" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.abs()=Bio.FSSP.FSSPAlignDict-class.html#abs"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-77', 'abs', 'link-77');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">simulated_fst</tt> <tt class="py-op">-</tt> <tt class="py-name">desired_fst</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">limit</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">parameter_fst</tt><tt class="py-op">,</tt> <tt class="py-name">max_run_fst</tt><tt class="py-op">,</tt> <tt class="py-name">min_run_fst</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">simulated_fst</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">desired_fst</tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">        <tt class="py-name">max_run_fst</tt> <tt class="py-op">=</tt> <tt class="py-name">parameter_fst</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">        <tt class="py-name">next_parameter_fst</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">min_run_fst</tt> <tt class="py-op">+</tt> <tt class="py-name">parameter_fst</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">        <tt class="py-name">min_run_fst</tt> <tt class="py-op">=</tt> <tt class="py-name">parameter_fst</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">        <tt class="py-name">next_parameter_fst</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">max_run_fst</tt> <tt class="py-op">+</tt> <tt class="py-name">parameter_fst</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">next_parameter_fst</tt><tt class="py-op">,</tt> <tt class="py-name">max_run_fst</tt><tt class="py-op">,</tt> <tt class="py-name">min_run_fst</tt> </tt>
</div><a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:05:02 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
